conftest.py

```diff
@@ -13,6 +14,9 @@
 )
 
 
+os.environ['WEB3_INFURA_PROJECT_ID'] = 'test-key'
```

Comment:
This seems... hacky. Let me know if there is a better way. Since I'm raising the error now instead of logging a warning, I can't run tests without this line

Reply:
You should be using the pytest `monkeypatch` fixture.  It has a mechanism for setting the variable in such a way that it reverts the change after the test/tests run.

Reply:
https://docs.pytest.org/en/latest/monkeypatch.html